x-pack/test/detection_engine_api_integration/security_and_spaces/tests/delete_rules_bulk.ts

```diff
@@ -249,6 +252,118 @@ export default ({ getService }: FtrProviderContext): void => {
           },
         ]);
       });
+
+      /**
+       * @deprecated Once the legacy notification system is removed, remove this test too.
+       */
+      it('should return the legacy action in the response body when it deletes a rule that has one', async () => {
+        // create an action
+        const { body: hookAction } = await supertest
+          .post('/api/actions/action')
+          .set('kbn-xsrf', 'true')
+          .send(getWebHookAction())
+          .expect(200);
+
+        // create a rule without actions
+        const createRuleBody = await createRule(supertest, log, getSimpleRule('rule-1'));
+
+        // Add a legacy rule action to the body of the rule
+        await createLegacyRuleAction(supertest, createRuleBody.id, hookAction.id);
+
+        // Test to ensure that we have exactly 1 legacy action by querying the Alerting client REST API directly
+        // See: https://www.elastic.co/guide/en/kibana/current/find-rules-api.html
+        // Note: We specifically query for both the filter of type "siem.notifications" and the "has_reference" to keep it very specific
+        await supertest
+          .get(`${BASE_ALERTING_API_PATH}/rules/_find`)
+          .query({
+            page: 1,
+            per_page: 10,
+            filter: 'alert.attributes.alertTypeId:(siem.notifications)',
+            has_reference: JSON.stringify({ id: createRuleBody.id, type: 'alert' }),
+          })
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+
+        // delete the rule with the legacy action
+        const { body } = await supertest
+          .delete(`${DETECTION_ENGINE_RULES_URL}/_bulk_delete`)
+          .send([{ id: createRuleBody.id }])
```

Comment:
Would it be worthwhile to check that the bulk API can migrate and delete _multiple_ rules, as opposed to just 1?

Reply:
Yeah, I can add an extra test for that situation as well.